doc/user/content/concepts/indexes.md

```diff
@@ -105,6 +105,100 @@ CREATE INDEX idx_on_my_view IN CLUSTER active_cluster ON my_view (...);
 
 ## Usage patterns
 
+### Index usage
+
+{{% important %}}
+Indexes are local to a cluster. Queries in a different cluster cannot use the indexes in another cluster.
+{{% /important %}}
+
+Unlike some other databases, Materialize can use an index to serve query
```

Comment:
Whether/why this is a good thing is a little obscure here, addressed in `doc/user/content/transform-data/optimization.md`. I don't know what level of duplication we want... maybe link there from here?